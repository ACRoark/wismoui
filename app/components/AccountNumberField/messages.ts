import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.account-number-field';

export default defineMessages({
  accountNumberFieldLabel: {
    id: `${scope}.account-number-field-label`,
    defaultMessage: 'Account Number',
  },
});
