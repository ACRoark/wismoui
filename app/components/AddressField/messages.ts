import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.address-field';

export default defineMessages({
  addressFieldLabel: {
    id: `${scope}.address-field-label`,
    defaultMessage: 'Address',
  },
});
