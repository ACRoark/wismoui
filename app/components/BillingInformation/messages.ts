import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.billing-information';

export default defineMessages({
  authenticatedUsers: {
    id: `${scope}.authenticated-users`,
  },
  billingInformationLabel: {
    id: `${scope}.billing-information-label`,
  },
  guestPayLink: {
    id: `${scope}.guest-pay-link`,
  },
  MIMO_START: {
    id: `${scope}.MIMO_START`,
  },
  MIMO_STOP: {
    id: `${scope}.MIMO_STOP`,
  },
  signInLink: {
    id: `${scope}.sign-in-link`,
  },
});
