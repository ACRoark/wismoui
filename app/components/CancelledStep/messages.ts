import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.cancelled-step';

export default defineMessages({
  MIMO_START: {
    id: `${scope}.MIMO_START`,
  },
  MIMO_STOP: {
    id: `${scope}.MIMO_STOP`,
  },
});
