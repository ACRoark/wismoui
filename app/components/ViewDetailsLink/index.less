@import (reference) '../../styles/themes/index.less';

.@{dte-prefix}-view-details-modal-content {
  display: block;
  text-align: left;

  .@{dte-prefix}-view-details-status-history-title {
    color: @dte-gray-700;
    display: block;
    font-weight: @dte-wismo-font-weight-semi-bold;
    padding: 20px 0;
    text-align: left;
  }

  .website {
    font-weight: @dte-wismo-font-weight-semi-bold;
    text-decoration: underline;
  }
}
