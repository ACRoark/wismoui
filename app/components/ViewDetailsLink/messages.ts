import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.view-details-link';

export default defineMessages({
  AD_title: {
    id: `${scope}.AD-title`,
  },
  AD_R975_description: {
    id: `${scope}.AD-R975-description`,
  },
  AD_R980_description: {
    id: `${scope}.AD-R980-description`,
  },
  AD_R981_description: {
    id: `${scope}.AD-R981-description`,
  },
  AD_R982_description: {
    id: `${scope}.AD-R982-description`,
  },
  AD_R983_description: {
    id: `${scope}.AD-R983-description`,
  },
  AN_R975_description: {
    id: `${scope}.AN-R975-description`,
  },
  AN_R980_description: {
    id: `${scope}.AN-R980-description`,
  },
  AN_R981_description: {
    id: `${scope}.AN-R981-description`,
  },
  AN_R982_description: {
    id: `${scope}.AN-R982-description`,
  },
  AN_R983_description: {
    id: `${scope}.AN-R983-description`,
  },
  AN_title: {
    id: `${scope}.AN-title`,
  },
  AP_R975_description: {
    id: `${scope}.AP-R975-description`,
  },
  AP_R980_description: {
    id: `${scope}.AP-R980-description`,
  },
  AP_R981_description: {
    id: `${scope}.AP-R981-description`,
  },
  AP_R982_description: {
    id: `${scope}.AP-R982-description`,
  },
  AP_R983_description: {
    id: `${scope}.AP-R983-description`,
  },
  AP_title: {
    id: `${scope}.AP-title`,
  },
  CANCELED_CGI_description : {
    id: `${scope}.CANCELED-CGI-description`,
  },
  CANCELED_CGI_title : {
    id: `${scope}.CANCELED-CGI-title`,
  },
  CANCELED_description : {
    id: `${scope}.CANCELED-description`,
  },
  CANCELED_INC_description : {
    id: `${scope}.CANCELED-INC-description`,
  },
  CANCELED_INC_title : {
    id: `${scope}.CANCELED-INC-title`,
  },
  CANCELED_title : {
    id: `${scope}.CANCELED-title`,
  },
  CL_R975_description: {
    id: `${scope}.CL-R975-description`,
  },
  CL_R980_description: {
    id: `${scope}.CL-R980-description`,
  },
  CL_R981_description: {
    id: `${scope}.CL-R981-description`,
  },
  CL_R982_description: {
    id: `${scope}.CL-R982-description`,
  },
  CL_R983_description: {
    id: `${scope}.CL-R983-description`,
  },
  CL_title: {
    id: `${scope}.CL-title`,
  },
  customerVerification: {
    id: `${scope}.customer-verification`,
  },
  DELAYED_description: {
    id: `${scope}.delayed-description`,
  },
  DELAYED_title: {
    id: `${scope}.delayed-title`,
  },
  DR_R975_description: {
    id: `${scope}.DR-R975-description`,
  },
  DR_R980_description: {
    id: `${scope}.DR-R980-description`,
  },
  DR_R981_description: {
    id: `${scope}.DR-R981-description`,
  },
  DR_R982_description: {
    id: `${scope}.DR-R982-description`,
  },
  DR_R983_description: {
    id: `${scope}.DR-R983-description`,
  },
  DR_title: {
    id: `${scope}.DR-title`,
  },
  ES_R975_description: {
    id: `${scope}.ES-R975-description`,
  },
  ES_R980_description: {
    id: `${scope}.ES-R980-description`,
  },
  ES_R981_description: {
    id: `${scope}.ES-R981-description`,
  },
  ES_R982_description: {
    id: `${scope}.ES-R982-description`,
  },
  ES_R983_description: {
    id: `${scope}.ES-R983-description`,
  },
  ES_R986_description: {
    id: `${scope}.ES-R986-description`,
  },
  ES_R987_description: {
    id: `${scope}.ES-R987-description`,
  },
  ES_title: {
    id: `${scope}.ES-title`,
  },
  FV_R975_description: {
    id: `${scope}.FV-R975-description`,
  },
  FV_R980_description: {
    id: `${scope}.FV-R980-description`,
  },
  FV_R981_description: {
    id: `${scope}.FV-R981-description`,
  },
  FV_R982_description: {
    id: `${scope}.FV-R982-description`,
  },
  FV_R983_description: {
    id: `${scope}.FV-R983-description`,
  },
  FV_title: {
    id: `${scope}.FV-title`,
  },
  FW_R975_description: {
    id: `${scope}.FW-R975-description`,
  },
  FW_R980_description: {
    id: `${scope}.FW-R980-description`,
  },
  FW_R981_description: {
    id: `${scope}.FW-R981-description`,
  },
  FW_R982_description: {
    id: `${scope}.FW-R982-description`,
  },
  FW_R983_description: {
    id: `${scope}.FW-R983-description`,
  },
  FW_title: {
    id: `${scope}.FW-title`,
  },
  LD_R983_description: {
    id: `${scope}.LD-R983-description`,
  },
  LD_title: {
    id: `${scope}.LD-title`,
  },
  PA_R984_description: {
    id: `${scope}.PA-R984-description`,
  },
  PA_R985_description: {
    id: `${scope}.PA-R985-description`,
  },
  PA_title: {
    id: `${scope}.PA-title`,
  },
  PM_R984_description: {
    id: `${scope}.PM-R984-description`,
  },
  PM_R985_description: {
    id: `${scope}.PM-R985-description`,
  },
  PM_title: {
    id: `${scope}.PM-title`,
  },
  PR_R975_description: {
    id: `${scope}.PR-R975-description`,
  },
  PR_R980_description: {
    id: `${scope}.PR-R980-description`,
  },
  PR_R981_description: {
    id: `${scope}.PR-R981-description`,
  },
  PR_R982_description: {
    id: `${scope}.PR-R982-description`,
  },
  PR_R983_description: {
    id: `${scope}.PR-R983-description`,
  },
  PR_title: {
    id: `${scope}.PR-title`,
  },
  RE_R975_description: {
    id: `${scope}.RE-R975-description`,
  },
  RE_R980_description: {
    id: `${scope}.RE-R980-description`,
  },
  RE_R981_description: {
    id: `${scope}.RE-R981-description`,
  },
  RE_R982_description: {
    id: `${scope}.RE-R982-description`,
  },
  RE_R983_description: {
    id: `${scope}.RE-R983-description`,
  },
  RE_title: {
    id: `${scope}.RE-title`,
  },
  RF_R975_description: {
    id: `${scope}.RF-R975-description`,
  },
  RF_R980_description: {
    id: `${scope}.RF-R980-description`,
  },
  RF_R981_description: {
    id: `${scope}.RF-R981-description`,
  },
  RF_R982_description: {
    id: `${scope}.RF-R982-description`,
  },
  RF_R983_description: {
    id: `${scope}.RF-R983-description`,
  },
  RF_R986_description: {
    id: `${scope}.RF-R986-description`,
  },
  RF_R987_description: {
    id: `${scope}.RF-R987-description`,
  },
  RF_title: {
    id: `${scope}.RF-title`,
  },
  SC_R975_description: {
    id: `${scope}.SC-R975-description`,
  },
  SC_R980_description: {
    id: `${scope}.SC-R980-description`,
  },
  SC_R981_description: {
    id: `${scope}.SC-R981-description`,
  },
  SC_R982_description: {
    id: `${scope}.SC-R982-description`,
  },
  SC_R983_description: {
    id: `${scope}.SC-R983-description`,
  },
  SC_title: {
    id: `${scope}.SC-title`,
  },
  status_history_title: {
    id: `${scope}.status-history-title`,
  },
  title: {
    id: `${scope}.title`,
  },
  website: {
    id: `${scope}.website`,
  },
  WP_R984_description: {
    id: `${scope}.WP-R984-description`,
  },
  WP_R985_description: {
    id: `${scope}.WP-R985-description`,
  },
  WP_title: {
    id: `${scope}.WP-title`,
  },
});
