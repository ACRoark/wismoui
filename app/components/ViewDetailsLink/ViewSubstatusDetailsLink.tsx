import * as _ from 'lodash';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import useModal from 'hooks/useModal';
import useThunkDispatch from 'hooks/useThunkDispatch';
import { trackButtonClick } from 'store/actions';

import IModalConfig from 'types/IModalConfig';

import './index.less';
import messages from './messages';

interface IViewDetailsLinkProps {
  className?: string;
  modal: IModalConfig;
}

const ViewSubstatusDetailsLink: React.FC<IViewDetailsLinkProps> = (
  props: IViewDetailsLinkProps,
): React.ReactElement => {
  const { className, modal } = props;

  const dispatch = useThunkDispatch();
  const { showModal } = useModal();

  const trackButtonAndOpenModal = (): void => {
    dispatch(trackButtonClick('view details link'));
    showModal(modal);
  };

  return (
    <Button className={className} onClick={trackButtonAndOpenModal} type="link">
      <FormattedMessage {...messages.title} />
    </Button>
  );
};

export default ViewSubstatusDetailsLink;
