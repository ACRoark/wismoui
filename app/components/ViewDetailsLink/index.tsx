import * as _ from 'lodash';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import useModal from 'hooks/useModal';
import useThunkDispatch from 'hooks/useThunkDispatch';
import { trackButtonClick } from 'store/actions';
import IModalConfig from 'types/IModalConfig';
import IOrderRequest from 'types/IOrderRequest';
import getServiceDate from 'utils/getServiceDate';

import getServiceOrderEventsModal from './getServiceOrderEventsModal';
import getVerificationEventsModal from './getVerificationEventsModal';

import './index.less';
import messages from './messages';

interface IViewDetailsLinkProps {
  className?: string;
  orderCreatedDate: string;
  serviceRequest: IOrderRequest;
}

const ViewDetailsLink: React.FC<IViewDetailsLinkProps> = (props: IViewDetailsLinkProps): React.ReactElement => {
  const { className, orderCreatedDate, serviceRequest } = props;
  const { serviceOrderEvents, verificationEvents } = serviceRequest;

  let serviceOrderEventStatus;

  if (serviceOrderEvents && serviceOrderEvents.length) {
    if (_.last(serviceOrderEvents).latestStatus) {
      serviceOrderEventStatus = _.last(serviceOrderEvents).latestStatus.serviceOrderEventStatus;
    }
  }

  const dispatch = useThunkDispatch();
  const { showModal } = useModal();
  const serviceDate = getServiceDate(serviceRequest) || orderCreatedDate;
  const isVerificationComplete = serviceOrderEvents && serviceOrderEvents.length;

  const emptyModal = { content: null, title: null };

  const ServiceOrderEventsModal = serviceOrderEventStatus
    ? getServiceOrderEventsModal(serviceOrderEventStatus)
    : emptyModal;

  const VerificationEventsModal =
    verificationEvents && verificationEvents.length
      ? getVerificationEventsModal(verificationEvents, serviceDate)
      : emptyModal;

  const modal: IModalConfig = isVerificationComplete ? ServiceOrderEventsModal : VerificationEventsModal;

  const trackButtonAndOpenModal = (): void => {
    dispatch(trackButtonClick('view details link'));
    showModal(modal);
  };

  return (
    <Button className={className} onClick={trackButtonAndOpenModal} type="link">
      <FormattedMessage {...messages.title} />
    </Button>
  );
};

export default ViewDetailsLink;
