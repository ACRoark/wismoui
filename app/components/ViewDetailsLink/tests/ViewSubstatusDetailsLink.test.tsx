
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Provider } from 'react-redux';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import useModal from 'hooks/useModal';
import {  ModalProvider } from 'providers/ModalProvider';
import { createMockStore, createSnapshotWithIntl } from 'testing/utils';
import IModalConfig from 'types/IModalConfig';

import ViewSubstatusDetailsLink from '../ViewSubstatusDetailsLink';

import { SUPPORTED_LOCALES } from 'locales';
import messages from '../messages';

jest.mock('hooks/useConfig');
jest.mock('hooks/useModal');

const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;
const mockModal = useModal as jest.MockedFunction<typeof useModal>;

const modalTitle = <FormattedMessage {...messages.DELAYED_title} />;

const modalContentForServiceOrderEvents = (
  <span className="dte-wismo-view-details-modal-content">
      <FormattedMessage
        {...messages.DELAYED_description}
        values={{
          phoneNumber: (
            <a className="phone-number" href="tel:+8004774747">
              800.477.4747
            </a>
          ),
          website: (
            <a href={'http://www.doo-dee-doo.com'}>
              <FormattedMessage {...messages.website} />
            </a>
          ),
        }}
      />
    </span>
);

const modalConfigForServiceOrderEvents: IModalConfig = {
  content: modalContentForServiceOrderEvents,
  title: modalTitle,
};

const expectComponentToMatchSnapshotForServiceOrderEvents = (language: string): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const tree = createSnapshotWithIntl(
    <Provider store={createMockStore()}>
       <ModalProvider {...modalConfigForServiceOrderEvents}>
         <ViewSubstatusDetailsLink modal={modalConfigForServiceOrderEvents} />
       </ModalProvider>
    </Provider>, language);

  expect(tree).toMatchSnapshot();
};

describe('<ViewDetailsLink2 />', (): void => {

  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {

    it(`should render the link when a valid serviceOrderEvents and serviceOrderEventStatus are provided (${language})`, (): void =>
      expectComponentToMatchSnapshotForServiceOrderEvents(language));
  });
});
