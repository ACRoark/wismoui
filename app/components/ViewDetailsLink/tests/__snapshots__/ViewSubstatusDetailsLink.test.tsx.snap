// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewDetailsLink2 /> should render the link when a valid serviceOrderEvents and serviceOrderEventStatus are provided (en) 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  onClick={[Function]}
>
  View Details
</button>
`;

exports[`<ViewDetailsLink2 /> should render the link when a valid serviceOrderEvents and serviceOrderEventStatus are provided (es) 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  onClick={[Function]}
>
  Ver Detalles
</button>
`;
